src/test/scala/mesosphere/marathon/health/MarathonHealthCheckManagerTest.scala

```diff
@@ -306,38 +305,5 @@ class MarathonHealthCheckManagerTest extends MarathonSpec with Logging {
     assert(hcManager.list(otherAppId) == otherHealthChecks)
   }
 
-  /**
-    * Captures the events send to the EventStream while the block is executing.
-    *
-    * For issue #2314 not only the end state is important. It is also important, which health checks
-    * were removed incorrectly and then readded afterwards.
-    */
-  private[this] def captureEvents(block: => Unit): Seq[MarathonEvent] = {
-    // yes, this is ugly. Since we only access it in the actor until it terminates, we do have
-    // the correct thread sync boundaries in place.
-
-    var capture = Vector.empty[MarathonEvent]
-    val captureEventsActor = actor {
-      new Act {
-        become {
-          case captureMe: MarathonEvent => capture :+= captureMe
-        }
-      }
-    }
-    eventStream.subscribe(captureEventsActor, classOf[MarathonEvent])
-    eventStream.subscribe(captureEventsActor, classOf[String])
-
-    try {
-      block
-    }
-    finally {
-      eventStream.unsubscribe(captureEventsActor)
-      captureEventsActor ! PoisonPill
-      val probe = TestProbe()
-      probe.watch(captureEventsActor)
-      probe.expectMsgClass(classOf[Terminated])
-    }
-
-    capture
-  }
+  def captureEvents = CaptureEvents(eventStream)
```

Comment:
I like your suggestion for the `capture` method. Making it less concise but clearer is fine with me. I'll do it in a follow up patch.

I like `new CaptureEvents` less than `CaptureEvents` and case classes give me the object apply method free-of-charge. I don't care much for unused auto-generated code but that's a personal preference. So what's your argument there? Is the use of a case class misleading or do you have performance gripes?